cpp/src/gandiva/gdv_function_stubs_test.cc

```diff
@@ -766,4 +766,64 @@ TEST(TestGdvFnStubs, TestCastVarbinaryFloat8) {
   ctx.Reset();
 }
 
+TEST(TestGdvFnStubs, TestMaskFirstN) {
+  gandiva::ExecutionContext ctx;
+  int64_t ctx_ptr = reinterpret_cast<int64_t>(&ctx);
+  int32_t out_len = 0;
+
+  const char* result = gdv_fn_mask_first_n(ctx_ptr, "0123456789", 10, 10, &out_len);
+  EXPECT_EQ("nnnnnnnnnn", std::string(result, out_len));
+
+  result = gdv_fn_mask_first_n(ctx_ptr, "ABCDEFGHIJKLMNOPQRSTUVWXYZ", 26, 26, &out_len);
+  EXPECT_EQ("XXXXXXXXXXXXXXXXXXXXXXXXXX", std::string(result, out_len));
+
+  result = gdv_fn_mask_first_n(ctx_ptr, "abcdefghijklmnopqrstuvwxyz", 26, 26, &out_len);
+  EXPECT_EQ("xxxxxxxxxxxxxxxxxxxxxxxxxx", std::string(result, out_len));
+
+  result = gdv_fn_mask_first_n(ctx_ptr, "aB-6", 4, 3, &out_len);
+  EXPECT_EQ("xX-6", std::string(result, out_len));
+
+  result = gdv_fn_mask_first_n(ctx_ptr, "aB-6", 4, -3, &out_len);
+  EXPECT_EQ("aB-6", std::string(result, out_len));
+
+  result = gdv_fn_mask_first_n(ctx_ptr, "aB-6", 4, 0, &out_len);
+  EXPECT_EQ("aB-6", std::string(result, out_len));
+
+  result = gdv_fn_mask_first_n(ctx_ptr, "ABcd-123456", 11, 6, &out_len);
+  EXPECT_EQ("XXxx-n23456", std::string(result, out_len));
+
+  result = gdv_fn_mask_first_n(ctx_ptr, "", 0, 6, &out_len);
+  EXPECT_EQ("", std::string(result, out_len));
+}
+
+TEST(TestGdvFnStubs, TestMaskLastN) {
+  gandiva::ExecutionContext ctx;
+  int64_t ctx_ptr = reinterpret_cast<int64_t>(&ctx);
+  int32_t out_len = 0;
+
+  const char* result = gdv_fn_mask_last_n(ctx_ptr, "0123456789", 10, 10, &out_len);
```

Comment:
Please add a test where input length < number of chars to mask

Reply:
Done